src/lib/server.ts

```diff
@@ -210,15 +215,27 @@ export class Client {
       if (err instanceof BaseError) {
         this.logger.error("Message error", err);
         const { errorCode, name, message, stack, ...args } = err;
-        return this.sendResultError(msg.messageId, errorCode, args);
+        if (this.schemaVersion < 32) {
+          this.sendResultZWaveError(
+            msg.messageId,
+            -1 as any,
+            `${errorCode}: ${message}`,
+          );
+        }
+        return this.sendResultError(msg.messageId, errorCode, message, args);
       }
       if (err instanceof ZWaveError) {
         this.logger.error("Z-Wave error", err);
         return this.sendResultZWaveError(msg.messageId, err.code, err.message);
       }
 
       this.logger.error("Unexpected error", err as Error);
-      this.sendResultError(msg.messageId, ErrorCode.unknownError, {});
+      this.sendResultError(
+        msg.messageId,
+        ErrorCode.unknownError,
+        (err as Error).toString(),
```

Comment:
Either
```suggestion
        err.message,
```
or
```suggestion
        err.stack,
```
depending on the level of detail you want to include.